src/utils/tree_node.C

```diff
@@ -485,40 +493,33 @@ const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_i
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element_in_children(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem* TreeNode<N>::find_element_in_children
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   libmesh_assert (!this->active());
 
-  unsigned int excluded_child = libMesh::invalid_uint;
+  std::vector<bool> searched_child(children.size(), false);
```

Comment:
If we want to avoid potential future threading problems, that precludes using `vector<anything>`!  Sure, `vector<bool>` is the worst case, since "write on one thread, read nearby element on another" can fail; but in any std::vector "append on one thread, read any element on another" can fail.  For that matter, "foo = foo + 1" can fail even when foo is a simple int.  AFAIK the only guaranteed way to avoid threading problems before-the-fact is by writing even the simplest code as nothing but "pure functions" - pass-by-value, no side effects, no non-const variables.

Now, *here*, we could change things to `vector<char>`.  I think "bool" is more self-documenting, but 7 bytes and an extra comment wouldn't be a big deal; let me know if you still want and I'll change it.  Let's just not go nuts with getting rid of vector<bool> elsewhere in the library until we need to; there's a few serial code paths where making an O(n_elem) vector 8 times bigger could hurt.